ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,453 @@
+      File input_vcfs
+      File input_vcf_indices
+      String output_basename
+    }
+     Int disk_size = size([input_vcfs, input_vcf_indices], "GB")
+  command <<<
+    bcftools norm -d both ~{input_vcfs} | bgzip -c > ~{output_basename}.vcf.gz
+    bcftools index -t ~{output_basename}.vcf.gz
+  >>>
+  runtime {
+    docker: "farjoun/impute:0.0.3-1504715575"
+    memory: "3 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+}
+
+task GenerateDataset {
+  input {
+    Array[File] input_vcfs
+    Array[File] input_vcf_indices
+    String output_vcf_basename
+   }
+
+   Int disk_size = size(input_vcfs, "GB") + size(input_vcf_indices, "GB") + 20
+
+    ### merge -> separate multiallelics -> remove all except SNP
+  command <<<
+    bcftools merge ~{sep=' ' input_vcfs} -O u | bcftools norm -m - -O z -o ~{output_vcf_basename}.vcf.gz 
+    # this shouldn't be necessary | awk 'BEGIN {FS="\t"}; {if($1 ~ /#/ || (($5=="A" || $5=="C" || $5=="G" || $5=="T") && ($4=="A" || $4=="C" || $4=="G" || $4=="T") && ($7 !~ "DUPE"))) print $0}' | bgzip -c > ~{output_vcf_basename}.vcf.gz
+    bcftools index -t ~{output_vcf_basename}.vcf.gz
+  >>>
+  runtime {
+    docker: "farjoun/impute:0.0.3-1504715575"
+    memory: "3 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_vcf = "~{output_vcf_basename}.vcf.gz"
+    File output_vcf_index = "~{output_vcf_basename}.vcf.gz.tbi"
+  }
+}
+
+task CalculateChromsomeLength {
+  input {
+    File ref_dict
+    Int chrom
+  }
+  command {
+    grep -P "SN:~{chrom}\t" ~{ref_dict} | sed 's/.*LN://' | sed 's/\t.*//'
+  }
+  runtime {
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk 100 HDD"
+    memory: "2 GB"
+  }
+  output {
+    Int chrom_length = read_int(stdout())
+  }
+}
+
+task ChunkVCF {
+  input {
+    Int start
+    Int end
+    String chrom
+    String basename
+    File vcf
+    File vcf_index
+    Int disk_size = 2*size([vcf, vcf_index], "GB")
+  }
+  command {
+    gatk SelectVariants -V ~{vcf} --select-type-to-include SNP --max-nocall-fraction 0.1 \
+    --restrict-alleles-to BIALLELIC -L ~{chrom}:~{start}-~{end} -O ~{basename}.vcf.gz
+  }
+  runtime {
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "8 GB"
+  }
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi" 
+  }
+}  
+
+task QConChunk {
+  input {
+    File vcf
+    File vcf_index
+    File panel_vcf
+    File panel_vcf_index
+    Int disk_size = size([vcf, vcf_index, panel_vcf, panel_vcf_index], "GB")
+  }
+  command <<<
+
+    ### IMPORTANT TODO: make sure ref and alts match between array and panel or are a simple mismatch
+ 
+    var_in_original=$(gatk CountVariants -V ~{vcf})
+    var_in_original=$(gatk CountVariants -V ~{vcf} -L ~{panel_vcf})
```

Comment:
I think this should be `var_in_reference`

Reply:
It is, Sarah corrected it already

Reply:
hmm, when I look at the latest commit it is still `var_in_original`... unless I am missing something.

I mean here in the git repo, not on a Terra snapshot.

Reply:
I see... (Terra version is corrected)